src/AppInstallerCLITests/PredefinedInstalledSource.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "TestCommon.h"
+#include <AppInstallerRepositorySource.h>
+#include <AppInstallerRuntime.h>
+#include <AppInstallerStrings.h>
+#include <Microsoft/PredefinedInstalledSourceFactory.h>
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+using namespace TestCommon;
+using namespace AppInstaller;
+using namespace AppInstaller::Repository;
+using namespace AppInstaller::Runtime;
+using namespace AppInstaller::Utility;
+
+using Factory = AppInstaller::Repository::Microsoft::PredefinedInstalledSourceFactory;
+
+std::shared_ptr<ISource> CreatePredefinedInstalledSource(Factory::Filter filter = Factory::Filter::None)
+{
+    SourceDetails details;
+    details.Type = Factory::Type();
+    details.Arg = Factory::FilterToString(filter);
+
+    TestProgress progress;
+
+    auto factory = Factory::Create();
+    return factory->Create(details, progress);
+}
+
+TEST_CASE("PredefinedInstalledSource_Create", "[installed][list]")

```

Comment:
>PredefinedInstalledSource_Create [](start = 11, length = 32)

Maybe redundant to make this a separate test since PredefinedInstalledSource_Search calls it anyway

Reply:
If this is failing, both tests will fail.  If this test passes, but the other does not, we know the issue is in Search (or a non-throwing failure to populate data).

Throwing everything in one test does not provide as much diagnostic data on failures as having multiple tests, even though they may repeat some of the same actions.

---
In reply to: [507576177](https://github.com/microsoft/winget-cli/pull/598#discussion_r507576177) [](ancestors = 507576177)